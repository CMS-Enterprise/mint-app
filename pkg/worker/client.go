package worker

import (
	"fmt"

	faktory "github.com/contribsys/faktory/client"
)

// NewClient returns a new faktory Client.
func NewClient() (*faktory.Client, error) {
	cl, err := faktory.Open()
	if err != nil {
		return nil, fmt.Errorf("error opening faktory client: %w", err)
	}
	return cl, nil
}
