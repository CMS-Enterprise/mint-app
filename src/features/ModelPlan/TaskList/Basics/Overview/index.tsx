import React, { Fragment, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useParams } from 'react-router-dom';
import {
  Button,
  Fieldset,
  Icon,
  Label,
  TextInput
} from '@trussworks/react-uswds';
import { Field, Form, Formik, FormikProps } from 'formik';
import {
  GetOverviewQuery,
  ModelType,
  TypedUpdateBasicsDocument,
  useGetOverviewQuery
} from 'gql/generated/graphql';

import AddNote from 'components/AddNote';
import AskAQuestion from 'components/AskAQuestion';
import Breadcrumbs, { BreadcrumbItemOptions } from 'components/Breadcrumbs';
import ConfirmLeave from 'components/ConfirmLeave';
import MutationErrorModal from 'components/MutationErrorModal';
import PageHeading from 'components/PageHeading';
import PageNumber from 'components/PageNumber';
import CheckboxField from 'components/CheckboxField';
import FieldGroup from 'components/FieldGroup';
import TextAreaField from 'components/TextAreaField';
import useHandleMutation from 'hooks/useHandleMutation';
import usePlanTranslation from 'hooks/usePlanTranslation';
import { getKeys } from 'types/translation';
import { NotFoundPartial } from 'features/NotFound';

type BasicsFormType = GetOverviewQuery['modelPlan']['basics'];

const Overview = () => {
  const { t: basicsT } = useTranslation('basics');
  const { t: basicsMiscT } = useTranslation('basicsMisc');
  const { t: miscellaneousT } = useTranslation('miscellaneous');

  const { modelType: modelTypeConfig } = usePlanTranslation('basics');

  const { modelID } = useParams<{ modelID: string }>();

  const formikRef = useRef<FormikProps<BasicsFormType>>(null);
  const history = useHistory();

  const { data, loading, error } = useGetOverviewQuery({
    variables: {
      id: modelID
    }
  });

  const { modelName } = data?.modelPlan || {};

  const {
    id,
    modelType,
    modelTypeOther,
    problem,
    goal,
    testInterventions,
    note
  } = (data?.modelPlan?.basics || {}) as BasicsFormType;

  const { mutationError } = useHandleMutation(TypedUpdateBasicsDocument, {
    id,
    formikRef
  });

  const initialValues: BasicsFormType = {
    __typename: 'PlanBasics',
    id: id ?? '',
    modelType: modelType ?? [],
    modelTypeOther: modelTypeOther ?? '',
    problem: problem ?? '',
    goal: goal ?? '',
    testInterventions: testInterventions ?? '',
    note: note ?? ''
  };

  if ((!loading && error) || (!loading && !data?.modelPlan)) {
    return <NotFoundPartial />;
  }

  return (
    <div data-testid="model-plan-overview">
      <MutationErrorModal
        isOpen={mutationError.isModalOpen}
        closeModal={() => mutationError.setIsModalOpen(false)}
        url={mutationError.destinationURL}
      />

      <Breadcrumbs
        items={[
          BreadcrumbItemOptions.HOME,
          BreadcrumbItemOptions.COLLABORATION_AREA,
          BreadcrumbItemOptions.TASK_LIST,
          BreadcrumbItemOptions.BASICS
        ]}
      />

      <PageHeading className="margin-top-4 margin-bottom-1">
        {basicsMiscT('heading')}
      </PageHeading>

      <p
        className="margin-top-0 margin-bottom-1 font-body-lg"
        data-testid="model-plan-name"
      >
        {miscellaneousT('for')} {modelName}
      </p>

      <p className="margin-bottom-2 font-body-md line-height-sans-4">
        {miscellaneousT('helpText')}
      </p>

      <AskAQuestion modelID={modelID} />

      <Formik
        initialValues={initialValues}
        onSubmit={() => {
          history.push(
            `/models/${modelID}/collaboration-area/task-list/basics/milestones`
          );
        }}
        enableReinitialize
        validateOnBlur={false}
        validateOnChange={false}
        validateOnMount={false}
        innerRef={formikRef}
      >
        {(formikProps: FormikProps<BasicsFormType>) => {
          const { dirty, handleSubmit, isValid, setErrors, values } =
            formikProps;

          return (
            <>
              <ConfirmLeave />

              <Form
                className="tablet:grid-col-6 margin-top-6"
                onSubmit={e => {
                  handleSubmit(e);
                  window.scrollTo(0, 0);
                }}
              >
                <Fieldset disabled={!!error || loading}>
                  <FieldGroup className="margin-top-4">
                    <Label htmlFor="modelType">
                      {basicsT('modelType.label')}
                    </Label>

                    <Fieldset>
                      {getKeys(modelTypeConfig.options).map(key => (
                        <Fragment key={key}>
                          <Field
                            as={CheckboxField}
                            id={`ModelType-${key}`}
                            name="modelType"
                            label={modelTypeConfig.options[key]}
                            value={key}
                            checked={values.modelType.includes(key)}
                          />
                        </Fragment>
                      ))}

                      {values.modelType?.includes(ModelType.OTHER) && (
                        <div className="margin-left-4">
                          <span>{basicsT('modelTypeOther.label')}</span>
                          <Field
                            as={TextInput}
                            id="ModelType-Other"
                            data-testid="ModelType-Other"
                            name="modelTypeOther"
                          />
                        </div>
                      )}
                    </Fieldset>
                  </FieldGroup>

                  <FieldGroup className="margin-top-4">
                    <Field
                      as={TextAreaField}
                      id="ModelType-Problem"
                      data-testid="ModelType-Problem"
                      name="problem"
                      label={basicsT('problem.label')}
                    />
                  </FieldGroup>

                  <FieldGroup className="margin-top-4">
                    <Field
                      as={TextAreaField}
                      id="ModelType-Goal"
                      name="goal"
                      hint={basicsT('goal.sublabel')}
                      label={basicsT('goal.label')}
                    />
                  </FieldGroup>

                  <FieldGroup className="margin-top-4">
                    <Field
                      as={TextAreaField}
                      id="ModelType-testInterventions"
                      name="testInterventions"
                      label={basicsT('testInterventions.label')}
                    />
                  </FieldGroup>

                  <AddNote id="ModelType-note" field="note" />

                  <div className="margin-top-6 margin-bottom-3">
                    <Button
                      type="button"
                      className="usa-button usa-button--outline margin-bottom-1"
                      onClick={() =>
                        history.push(
                          `/models/${modelID}/collaboration-area/task-list/basics`
                        )
                      }
                    >
                      {miscellaneousT('back')}
                    </Button>

                    <Button
                      type="submit"
                      disabled={!(dirty || isValid)}
                      className=""
                      onClick={() => setErrors({})}
                    >
                      {miscellaneousT('next')}
                    </Button>
                  </div>
                  <Button
                    type="button"
                    className="usa-button usa-button--unstyled"
                    onClick={() =>
                      history.push(
                        `/models/${modelID}/collaboration-area/task-list`
                      )
                    }
                  >
                    <Icon.ArrowBack className="margin-right-1" aria-hidden />

                    {miscellaneousT('saveAndReturn')}
                  </Button>
                </Fieldset>
              </Form>
            </>
          );
        }}
      </Formik>

      <PageNumber currentPage={2} totalPages={3} className="margin-bottom-10" />
    </div>
  );
};

export default Overview;
