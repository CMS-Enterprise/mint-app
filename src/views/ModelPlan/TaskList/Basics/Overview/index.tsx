import React, { useRef, useState } from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Link, useHistory, useParams } from 'react-router-dom';
import { useMutation, useQuery } from '@apollo/client';
import {
  Breadcrumb,
  BreadcrumbBar,
  BreadcrumbLink,
  Button,
  Fieldset,
  IconAdd,
  IconArrowBack,
  Label,
  Radio
} from '@trussworks/react-uswds';
import { Field, Form, Formik, FormikProps } from 'formik';

import MainContent from 'components/MainContent';
import PageHeading from 'components/PageHeading';
import PageNumber from 'components/PageNumber';
import AutoSave from 'components/shared/AutoSave';
import { ErrorAlert, ErrorAlertMessage } from 'components/shared/ErrorAlert';
import FieldErrorMsg from 'components/shared/FieldErrorMsg';
import FieldGroup from 'components/shared/FieldGroup';
import TextAreaField from 'components/shared/TextAreaField';
import GetModelPlanQuery from 'queries/GetModelPlanQuery';
import {
  GetModelPlan,
  GetModelPlanVariables
} from 'queries/types/GetModelPlan';
import { UpdatePlanBasics as UpdatePlanBasicsType } from 'queries/types/UpdatePlanBasics';
import UpdatePlanBasics from 'queries/UpdatePlanBasics';
import flattenErrors from 'utils/flattenErrors';
import planBasicsSchema from 'validations/planBasics';

interface PlanBasicsOverviewTypes {
  modelType: string | null;
  problem: string;
  goal: string;
  testInterventions: string;
  note: string;
}

const Overview = () => {
  const { t } = useTranslation('basics');
  const { t: h } = useTranslation('draftModelPlan');
  const { modelId } = useParams<{ modelId: string }>();

  const formikRef = useRef<FormikProps<PlanBasicsOverviewTypes>>(null);
  const history = useHistory();
  const [hasAdditionalNote, setHasAdditionalNote] = useState(false);

  const { data } = useQuery<GetModelPlan, GetModelPlanVariables>(
    GetModelPlanQuery,
    {
      variables: {
        id: modelId
      }
    }
  );

  const { modelName } = data?.modelPlan || {};
  const { id, modelType, problem, goal, testInventions, note } =
    data?.modelPlan?.basics || {};

  const [update] = useMutation<UpdatePlanBasicsType>(UpdatePlanBasics);

  const initialValues = {
    modelType: modelType ?? null,
    problem: problem ?? '',
    goal: goal ?? '',
    testInterventions: testInventions ?? '',
    note: note ?? ''
  };

  const handleFormSubmit = (
    formikValues: PlanBasicsOverviewTypes,
    redirect?: 'next' | 'back' | 'task-list'
  ) => {
    update({
      variables: {
        id,
        changes: {
          modelType: formikValues.modelType,
          problem: formikValues.problem,
          goal: formikValues.goal,
          testInventions: formikValues.testInterventions,
          note: formikValues.note
        }
      }
    })
      .then(response => {
        if (!response?.errors) {
          if (redirect === 'next') {
            history.push(`/models/${modelId}/task-list/basics/milestones`);
          } else if (redirect === 'back') {
            history.push(`/models/${modelId}/task-list/basics`);
          } else if (redirect === 'task-list') {
            history.push(`/models/${modelId}/task-list/`);
          }
        }
      })
      .catch(errors => {
        formikRef?.current?.setErrors(errors);
      });
  };

  return (
    <MainContent className="margin-bottom-5">
      <div className="grid-container">
        <div className="tablet:grid-col-12">
          <BreadcrumbBar variant="wrap">
            <Breadcrumb>
              <BreadcrumbLink asCustom={Link} to="/">
                <span>{h('home')}</span>
              </BreadcrumbLink>
            </Breadcrumb>
            <Breadcrumb>
              <BreadcrumbLink
                asCustom={Link}
                to={`/models/${modelId}/task-list/`}
              >
                <span>{h('tasklistBreadcrumb')}</span>
              </BreadcrumbLink>
            </Breadcrumb>
            <Breadcrumb current>{t('breadcrumb')}</Breadcrumb>
          </BreadcrumbBar>
          <PageHeading className="margin-top-4 margin-bottom-1">
            {t('heading')}
          </PageHeading>
          <p
            className="margin-top-0 margin-bottom-1 font-body-lg"
            data-testid="model-plan-name"
          >
            <Trans i18nKey="modelPlanTaskList:subheading">
              indexZero {modelName} indexTwo
            </Trans>
          </p>
          <p className="margin-bottom-2 font-body-md line-height-sans-4">
            {t('helpText')}
          </p>

          <Formik
            initialValues={initialValues}
            onSubmit={values => {
              handleFormSubmit(values, 'next');
            }}
            enableReinitialize
            validationSchema={planBasicsSchema.pageTwoSchema}
            validateOnBlur={false}
            validateOnChange={false}
            validateOnMount={false}
            innerRef={formikRef}
          >
            {(formikProps: FormikProps<PlanBasicsOverviewTypes>) => {
              const {
                dirty,
                errors,
                handleSubmit,
                isValid,
                setErrors,
                values
              } = formikProps;
              const flatErrors = flattenErrors(errors);
              return (
                <>
                  {Object.keys(errors).length > 0 && (
                    <ErrorAlert
                      testId="formik-validation-errors"
                      classNames="margin-top-3"
                      heading="Please check and fix the following"
                    >
                      {Object.keys(flatErrors).map(key => {
                        return (
                          <ErrorAlertMessage
                            key={`Error.${key}`}
                            errorKey={key}
                            message={flatErrors[key]}
                          />
                        );
                      })}
                    </ErrorAlert>
                  )}
                  <Form
                    className="tablet:grid-col-6"
                    onSubmit={e => {
                      handleSubmit(e);
                      window.scrollTo(0, 0);
                    }}
                  >
                    <FieldGroup
                      scrollElement="modelType"
                      error={!!flatErrors.modelType}
                      className="margin-top-4"
                    >
                      <Label htmlFor="modelType">{t('modelType')}</Label>
                      <FieldErrorMsg>{flatErrors.modelType}</FieldErrorMsg>
                      <Fieldset>
                        <Field
                          as={Radio}
                          id="ModelType-Voluntary"
                          name="modelType"
                          label={t('voluntary')}
                          value="VOLUNTARY"
                          checked={values.modelType === 'VOLUNTARY'}
                        />
                        <Field
                          as={Radio}
                          id="ModelType-Mandatory"
                          name="modelType"
                          label={t('Mandatory')}
                          value="MANDATORY"
                          checked={values.modelType === 'MANDATORY'}
                        />
                      </Fieldset>
                    </FieldGroup>

                    <FieldGroup
                      scrollElement="problem"
                      error={!!flatErrors.problem}
                      className="margin-top-4"
                    >
                      <Field
                        as={TextAreaField}
                        className="height-15"
                        error={flatErrors.problem}
                        id="ModelType-Problem"
                        name="problem"
                        label={t('problem')}
                      />
                    </FieldGroup>

                    <FieldGroup
                      scrollElement="goal"
                      error={!!flatErrors.goal}
                      className="margin-top-4"
                    >
                      <Field
                        as={TextAreaField}
                        className="height-15"
                        error={flatErrors.goal}
                        id="ModelType-Goal"
                        name="goal"
                        hint={t('goalHelp')}
                        label={t('goal')}
                      />
                    </FieldGroup>

                    <FieldGroup
                      scrollElement="testInterventions"
                      error={!!flatErrors.testInterventions}
                      className="margin-top-4"
                    >
                      <Field
                        as={TextAreaField}
                        className="height-15"
                        error={flatErrors.testInterventions}
                        id="ModelType-testInterventions"
                        name="testInterventions"
                        label={t('testInterventions')}
                      />
                    </FieldGroup>

                    <Button
                      type="button"
                      className="usa-button usa-button--unstyled margin-top-4"
                      onClick={() => setHasAdditionalNote(true)}
                    >
                      <IconAdd className="margin-right-1" aria-hidden />
                      {h('additionalNote')}
                    </Button>

                    {hasAdditionalNote && (
                      <FieldGroup className="margin-top-4">
                        <Field
                          as={TextAreaField}
                          className="height-15"
                          id="ModelType-note"
                          name="note"
                          label={t('Notes')}
                        />
                      </FieldGroup>
                    )}

                    <div className="margin-top-6 margin-bottom-3">
                      <Button
                        type="button"
                        className="usa-button usa-button--outline margin-bottom-1"
                        onClick={() =>
                          history.push(`/models/${modelId}/task-list/basics`)
                        }
                      >
                        {h('back')}
                      </Button>
                      <Button
                        type="submit"
                        disabled={!(dirty || isValid)}
                        className=""
                        onClick={() => setErrors({})}
                      >
                        {h('next')}
                      </Button>
                    </div>
                    <Button
                      type="button"
                      className="usa-button usa-button--unstyled"
                      onClick={() => handleFormSubmit(values, 'task-list')}
                    >
                      <IconArrowBack className="margin-right-1" aria-hidden />
                      {h('saveAndReturn')}
                    </Button>
                  </Form>
                  <AutoSave
                    values={values}
                    onSave={() => {
                      if (
                        Object.keys(formikRef.current!.touched).length !== 0
                      ) {
                        handleFormSubmit(formikRef.current!.values);
                      }
                    }}
                    debounceDelay={3000}
                  />
                </>
              );
            }}
          </Formik>
        </div>
        <PageNumber
          currentPage={2}
          totalPages={3}
          className="margin-bottom-10"
        />
      </div>
    </MainContent>
  );
};

export default Overview;
