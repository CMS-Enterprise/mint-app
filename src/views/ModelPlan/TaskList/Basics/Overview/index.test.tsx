import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, waitFor } from '@testing-library/react';
import {
  GetOverviewDocument,
  GetOverviewQuery,
  ModelType
} from 'gql/gen/graphql';

import Overview from './index';

type GetOverviewType = GetOverviewQuery['modelPlan']['basics'];

const overviewMockData: GetOverviewType = {
  __typename: 'PlanBasics',
  id: '123',
  modelType: ModelType.MANDATORY,
  problem: 'My problem',
  goal: 'A goal',
  testInterventions: 'Intervention',
  note: 'Test note'
};

const mocks = [
  {
    request: {
      query: GetOverviewDocument,
      variables: { id: 'f11eb129-2c80-4080-9440-439cbe1a286f' }
    },
    result: {
      data: {
        modelPlan: {
          __typename: 'ModelPlan',
          id: 'f11eb129-2c80-4080-9440-439cbe1a286f',
          modelName: 'My excellent plan that I just initiated',
          basics: overviewMockData
        }
      }
    }
  }
];

describe('Basics overview page', () => {
  it('renders without errors and matches snapshot', async () => {
    const { asFragment, getByTestId } = render(
      <MemoryRouter
        initialEntries={[
          '/models/f11eb129-2c80-4080-9440-439cbe1a286f/task-list/overview'
        ]}
      >
        <MockedProvider mocks={mocks} addTypename={false} showWarnings>
          <Route path="/models/:modelID/task-list/overview">
            <Overview />
          </Route>
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getByTestId('ModelType-Problem')).toHaveValue('My problem');
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
