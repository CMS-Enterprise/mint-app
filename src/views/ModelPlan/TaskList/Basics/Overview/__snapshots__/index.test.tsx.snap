// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Basics overview page > renders without errors and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="model-plan-overview"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      >
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-breadcrumb__link"
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-breadcrumb__link"
            href="/models/f11eb129-2c80-4080-9440-439cbe1a286f/task-list/"
          >
            <span>
              Model Plan task list
            </span>
          </a>
        </li>
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
        >
          Model basics
        </li>
      </ol>
    </nav>
    <h1
      aria-live="polite"
      class="mint-h1 margin-top-4 margin-bottom-1"
      tabindex="-1"
    >
      Model basics
    </h1>
    <p
      class="margin-top-0 margin-bottom-1 font-body-lg"
      data-testid="model-plan-name"
    >
      for My excellent plan that I just initiated
    </p>
    <p
      class="margin-bottom-2 font-body-md line-height-sans-4"
    >
      If there's a question or field that is not applicable to your model or you don't currently know the answer, you may leave it blank. If you need help, ask a question using the link below.
    </p>
    <div
      class="padding-2 bg-primary-lighter"
    >
      <div
        class="display-flex"
        data-testid="ask-a-question"
      >
        <svg
          class="usa-icon text-primary margin-right-1"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
          />
        </svg>
        <button
          class="usa-button usa-button--unstyled"
          data-testid="ask-a-question-button"
          type="button"
        >
          Start a discussion
        </button>
      </div>
    </div>
    <form
      action="#"
      class="tablet:grid-col-6 margin-top-6"
    >
      <fieldset
        class="usa-fieldset"
        data-testid="fieldset"
      >
        <div
          class="usa-form-group margin-top-4"
          data-scroll="modelType"
        >
          <label
            class="usa-label"
            data-testid="label"
            for="modelType"
          >
            Model Type
          </label>
          <fieldset
            class="usa-fieldset"
            data-testid="fieldset"
          >
            <div
              class="usa-radio"
              data-testid="radio"
            >
              <input
                class="usa-radio__input"
                id="ModelType-Voluntary"
                name="modelType"
                type="radio"
                value="VOLUNTARY"
              />
              <label
                class="usa-radio__label"
                for="ModelType-Voluntary"
              >
                Voluntary
              </label>
            </div>
            <div
              class="usa-radio"
              data-testid="radio"
            >
              <input
                class="usa-radio__input"
                id="ModelType-Mandatory"
                name="modelType"
                type="radio"
                value="MANDATORY"
              />
              <label
                class="usa-radio__label"
                for="ModelType-Mandatory"
              >
                Mandatory
              </label>
            </div>
          </fieldset>
        </div>
        <div
          class="usa-form-group margin-top-4"
          data-scroll="problem"
        >
          <label
            class="usa-label"
            for="ModelType-Problem"
            id="label-ModelType-Problem"
          >
            Problem statement
          </label>
          <textarea
            aria-labelledby="label-ModelType-Problem hint-ModelType-Problem"
            class="usa-textarea"
            data-testid="ModelType-Problem"
            id="ModelType-Problem"
            name="problem"
          >
            My problem
          </textarea>
        </div>
        <div
          class="usa-form-group margin-top-4"
          data-scroll="goal"
        >
          <label
            class="usa-label"
            for="ModelType-Goal"
            id="label-ModelType-Goal"
          >
            Goal
          </label>
          <span
            class="usa-hint display-block text-normal"
            id="hint-ModelType-Goal"
          >
            Please include the high level goal of the program and a description of the project.
          </span>
          <textarea
            aria-labelledby="label-ModelType-Goal hint-ModelType-Goal"
            class="usa-textarea"
            id="ModelType-Goal"
            name="goal"
          >
            A goal
          </textarea>
        </div>
        <div
          class="usa-form-group margin-top-4"
          data-scroll="testInterventions"
        >
          <label
            class="usa-label"
            for="ModelType-testInterventions"
            id="label-ModelType-testInterventions"
          >
            Test Interventions
          </label>
          <textarea
            aria-labelledby="label-ModelType-testInterventions hint-ModelType-testInterventions"
            class="usa-textarea"
            id="ModelType-testInterventions"
            name="testInterventions"
          >
            Intervention
          </textarea>
        </div>
        <div
          class="margin-top-4 margin-bottom-8"
        >
          <div
            class="usa-form-group"
          >
            <label
              class="usa-label"
              for="ModelType-note"
              id="label-ModelType-note"
            >
              Notes
            </label>
            <textarea
              aria-labelledby="label-ModelType-note hint-ModelType-note"
              class="usa-textarea"
              data-testid="ModelType-note"
              id="ModelType-note"
              name="note"
            >
              Test note
            </textarea>
          </div>
        </div>
        <div
          class="margin-top-6 margin-bottom-3"
        >
          <button
            class="usa-button usa-button usa-button--outline margin-bottom-1"
            data-testid="button"
            type="button"
          >
            Back
          </button>
          <button
            class="usa-button"
            data-testid="button"
            type="submit"
          >
            Next
          </button>
        </div>
        <button
          class="usa-button usa-button usa-button--unstyled"
          data-testid="button"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="usa-icon margin-right-1"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Save and return to task list
        </button>
      </fieldset>
    </form>
    <div
      class="mint-page-number margin-bottom-10"
    >
      <span
        class="mint-page-number__page-num"
      >
        Page 2 of 3
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Model Plan Documents page > matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="model-plan-overview"
  >
    <nav
      aria-label="Breadcrumbs"
      class="usa-breadcrumb usa-breadcrumb--wrap"
    >
      <ol
        class="usa-breadcrumb__list"
      >
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-breadcrumb__link"
            href="/"
          >
            <span>
              Home
            </span>
          </a>
        </li>
        <li
          class="usa-breadcrumb__list-item"
        >
          <a
            class="usa-breadcrumb__link"
            href="/models/f11eb129-2c80-4080-9440-439cbe1a286f/task-list/"
          >
            <span>
              Model Plan task list
            </span>
          </a>
        </li>
        <li
          aria-current="page"
          class="usa-breadcrumb__list-item usa-current"
        >
          Model basics
        </li>
      </ol>
    </nav>
    <h1
      aria-live="polite"
      class="mint-h1 margin-top-4 margin-bottom-1"
      tabindex="-1"
    >
      Model basics
    </h1>
    <p
      class="margin-top-0 margin-bottom-1 font-body-lg"
      data-testid="model-plan-name"
    >
      for 
    </p>
    <p
      class="margin-bottom-2 font-body-md line-height-sans-4"
    >
      If there's a question or field that is not applicable to your model or you don't currently know the answer, you may leave it blank. If you need help, ask a question using the link below.
    </p>
    <div
      class="padding-2 bg-primary-lighter"
    >
      <div
        class="display-flex"
        data-testid="ask-a-question"
      >
        <svg
          class="usa-icon text-primary margin-right-1"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
          />
        </svg>
        <button
          class="usa-button usa-button--unstyled"
          data-testid="ask-a-question-button"
          type="button"
        >
          Start a discussion
        </button>
      </div>
    </div>
    <form
      action="#"
      class="tablet:grid-col-6 margin-top-6"
    >
      <fieldset
        class="usa-fieldset"
        data-testid="fieldset"
        disabled=""
      >
        <div
          class="usa-form-group margin-top-4"
          data-scroll="modelType"
        >
          <label
            class="usa-label"
            data-testid="label"
            for="modelType"
          >
            Model Type
          </label>
          <fieldset
            class="usa-fieldset"
            data-testid="fieldset"
          >
            <div
              class="usa-radio"
              data-testid="radio"
            >
              <input
                class="usa-radio__input"
                id="ModelType-Voluntary"
                name="modelType"
                type="radio"
                value="VOLUNTARY"
              />
              <label
                class="usa-radio__label"
                for="ModelType-Voluntary"
              >
                Voluntary
              </label>
            </div>
            <div
              class="usa-radio"
              data-testid="radio"
            >
              <input
                class="usa-radio__input"
                id="ModelType-Mandatory"
                name="modelType"
                type="radio"
                value="MANDATORY"
              />
              <label
                class="usa-radio__label"
                for="ModelType-Mandatory"
              >
                Mandatory
              </label>
            </div>
          </fieldset>
        </div>
        <div
          class="usa-form-group margin-top-4"
          data-scroll="problem"
        >
          <label
            class="usa-label"
            for="ModelType-Problem"
            id="label-ModelType-Problem"
          >
            Problem statement
          </label>
          <textarea
            aria-labelledby="label-ModelType-Problem hint-ModelType-Problem"
            class="usa-textarea"
            id="ModelType-Problem"
            name="problem"
          />
        </div>
        <div
          class="usa-form-group margin-top-4"
          data-scroll="goal"
        >
          <label
            class="usa-label"
            for="ModelType-Goal"
            id="label-ModelType-Goal"
          >
            Goal
          </label>
          <span
            class="usa-hint display-block text-normal"
            id="hint-ModelType-Goal"
          >
            Please include the high level goal of the program and a description of the project.
          </span>
          <textarea
            aria-labelledby="label-ModelType-Goal hint-ModelType-Goal"
            class="usa-textarea"
            id="ModelType-Goal"
            name="goal"
          />
        </div>
        <div
          class="usa-form-group margin-top-4"
          data-scroll="testInterventions"
        >
          <label
            class="usa-label"
            for="ModelType-testInterventions"
            id="label-ModelType-testInterventions"
          >
            Test Interventions
          </label>
          <textarea
            aria-labelledby="label-ModelType-testInterventions hint-ModelType-testInterventions"
            class="usa-textarea"
            id="ModelType-testInterventions"
            name="testInterventions"
          />
        </div>
        <div
          class="margin-top-4 margin-bottom-8"
        >
          <button
            class="usa-button usa-button usa-button--unstyled"
            data-testid="add-note-toggle"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="usa-icon margin-right-1"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
            Add an additional note
          </button>
        </div>
        <div
          class="margin-top-6 margin-bottom-3"
        >
          <button
            class="usa-button usa-button usa-button--outline margin-bottom-1"
            data-testid="button"
            type="button"
          >
            Back
          </button>
          <button
            class="usa-button"
            data-testid="button"
            type="submit"
          >
            Next
          </button>
        </div>
        <button
          class="usa-button usa-button usa-button--unstyled"
          data-testid="button"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="usa-icon margin-right-1"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Save and return to task list
        </button>
      </fieldset>
    </form>
    <div
      class="mint-page-number margin-bottom-10"
    >
      <span
        class="mint-page-number__page-num"
      >
        Page 2 of 3
      </span>
    </div>
  </div>
</DocumentFragment>
`;
